scripts/openapi/marketo/metadata/metadata.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/amp-labs/connectors/tools/scrapper"
+	"github.com/getkin/kin-openapi/openapi2"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+func main() {
+	//  read the definitions in the specificatios files.
+	def, docA, err := constructDefinitions("./scripts/openapi/marketo/metadata/asset.json", 5) //nolint:gomnd
+	if err != nil {
+		panic(err)
+	}
+
+	ldef, docL, err := constructDefinitions("./scripts/openapi/marketo/metadata/mapi.json", 4) //nolint:gomnd
+	if err != nil {
+		panic(err)
+	}
+
+	// Initializes an empty ObjectMetadata variable
+	objectMetadata := make(map[string]scrapper.ObjectMetadata)
+
+	// Add Lead metadata details
+	objectMetadata, err = generateMetadata(ldef, docL, objectMetadata)
+	if err != nil {
+		panic(err)
+	}
+
+	// Adds Assets Metadata details to the same variable declared above.
+	objectMetadata, err = generateMetadata(def, docA, objectMetadata)
+	if err != nil {
+		panic(err)
+	}
+
+	// wrap objectMetadata in `data` to not break the scrapper API.
+	data := map[string]any{
+		"data": objectMetadata,
+	}
+
+	mb, err := json.Marshal(data)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a `schemas.json` file and Adds the metadata details to the file.
+	if err = writefile(mb); err != nil {
+		panic(err)
+	}
+}
+
+func writefile(b []byte) error {
+	f, err := os.Create("schemas.json")
```

Comment:
The file manager from scrapper package which you have already declared (maybe package should be called differently as it no longer makes sense, before it was scrapper only).
```
FileManager = scrapper.NewMetadataFileManager(schemas, fileconv.NewSiblingFileLocator())
```
This filemanager has utility methods like `SaveSchemas()`. And it will save it to the neighbouring schema.json file.
This method accepts `scrapper.NewObjectMetadataResult()` data structure, which has convenient `Add` method this can be used to add schemas from one file then from the other and finally save to `schemas.json`.

@Jkarage If you have any questions I can help you, dont hesitate to ask!

Reply:
@Cobalt0s yeah, I saw the tool as well. The reason I decided to create the script is as the wrapper `api3` uses the OpenAPI v3 parser. And the marketo files needs v2. We can look more into it if they are full compatible and update this script. Or Update the wrapper to also include the v2 parser.

Reply:
Well, that is really unfortunate... I guess we can later think about having api3 and api2 with some common core code. I wonder how many more connectors will have v2.
I guess only `scrapper.NewMetadataFileManager` could be of use.